Gems/Atom/RPI/Code/Source/RPI.Public/Pass/PassUtils.cpp

```diff
@@ -37,6 +37,49 @@ namespace AZ
                 return passData;
             }
 
+            AZStd::shared_ptr<PassData> GetPassDataPtr(const PassDescriptor& descriptor)
+            {
+                AZStd::shared_ptr<PassData> passData = nullptr;
+
+                if (descriptor.m_passRequest != nullptr)
+                {
+                    passData = descriptor.m_passRequest->m_passData;
+                }
+                if (passData == nullptr && descriptor.m_passTemplate != nullptr)
+                {
+                    passData = descriptor.m_passTemplate->m_passData;
+                }
+                if (passData == nullptr)
+                {
+                    passData = descriptor.m_passData;
+                }
+                return passData;
+            }
+
+            void ExtractPipelineGlobalConnections(const AZStd::shared_ptr<PassData>& passData, PipelineGlobalConnectionList& outList)
+            {
+                for (const PipelineGlobalConnection& connection : passData->m_pipelineGlobalConnections)
+                {
+                    outList.push_back(connection);
+                }
+            }
+
+            void ExtractPipelineGlobalConnections(const PassDescriptor& descriptor, PipelineGlobalConnectionList& outList)
```

Comment:
The logic of getting passData (use priority) is different than getting pipeline global connections (combined). 

I think this is not be a good idea. People won't expect this unless they actually look at the implementation code. 

I would rather we use the same logic as getting passData

So the function should be 
ExtractPipelineGlobalConnections(GetPassDataPtr(descrptor), outList).